tests/test_cases/test_array/test_array.py

```diff
@@ -400,7 +403,9 @@ async def test_direct_constant_indexing(dut):
     _check_type(tlog, dut.const_cmplx[1].b[1], ConstantObject)
 
 
-@cocotb.test()
+@cocotb.test(
+    # Icarus 10.3 doesn't support bit-selects, see https://github.com/steveicarus/iverilog/issues/323
+    expect_error=IndexError if cocotb.SIM_NAME.lower().startswith("icarus") else False)
```

Comment:
Icarus 11.0 has been released and supports bit selects. Should we predicate it on the version number in `cocotb.SIM_VERSION`?

Reply:
Let's close #1987 in the next days one way or the other and use it here.

Reply:
I fixed it up a bit now, but the `sig_t2` thing still needs to be checked.

Reply:
Even in 12.0, `sig_t2` is not seen in Icarus, but I wonder if it is a matter of optimizing those away see also https://github.com/steveicarus/iverilog/issues/322). There is `sig_t1` too which is just a 1-dim vector wire that is also not seen.
I'll try using those wires for something in HDL and check more.

Reply:
That was it indeed. Current problem: https://github.com/cocotb/cocotb/runs/1542442627?check_suite_focus=true#step:22:656 (ping @garmin-mjames)

Reply:
Icarus flattens multi-dimensional arrays into a single dimension, so a 4x4 matrix becomes 1x16.

Reply:
Thanks, something to be documented and a special case to be introduced.